src/core/Pattern_match.ml

```diff
@@ -129,8 +129,20 @@ type t = {
      the override is applied on top of the default and only changes the fields
      present in the override. *)
   metadata_override : JSON.t option;
+  dependency_match_data : dependency_match_data;
 }
 
+and dependency_match_data =
+  (* Rule had both code patterns and dependency patterns, got matches on *both*, the Pattern Match is in code, annotated with this dependency match *)
+  | CodePMwith of dependency_match
+  (* Rule had dependency patterns, they matched, the Pattern Match is in a lockfile *)
+  (* So the range_loc of the Dependency.t in this dependency_match should be *the same* as the range_loc in the PatternMatch.t *)
+  | LockfileOnlyPM of dependency_match
+  (* Rule had no dependency patterns, this is a regular code pattern match *)
+  | CodeOnlyPM
```

Comment:
maybe remove this case and make the type an option in the dependency field above.